### ISSUE TITLE
Anchor free models

### ISSUE THREAD

**Issue by bhack** [keras-team/keras-cv#1]
Please evaluate to extract common components/utils that could support Anchor free models:
https://github.com/tensorflow/hub/issues/424

See also: 
https://github.com/google/mediapipe/issues/495#issuecomment-616721458
https://github.com/google-coral/edgetpu/issues/51

**Comment by SunPaz** [comment]
Very naive questions here: 
1) Why would you like to implement an anchor free model in an industrial context ?
AFAIK anchor model are widely used for label inspection as they give a very quick and reliable result.
2) Have you considered adressing vision inspection problems (detect defects on products) or labeling only (read and check label information) ? 

I would be very happy to discuss further on this subject, and maybe try to participate on this project, as I m currently trying to implement a (very humble) anchor based "old school" vision control library.

**Comment by tanzhenyu** [comment]
Anchor free models are probably not in our immediate roadmaps yet. @fchollet WDYT?

**Comment by bhack** [comment]
@tanzhenyu At least it would have a bit of a way to diversify keras-cv compared to the model garden since almost all official anchor free papers implementations are in pytorch

**Comment by tanzhenyu** [comment]
> @tanzhenyu At least it would have a bit of a way to diversify keras-cv compared to the model garden since almost all official anchor free papers implementations are in pytorch.
> 
> EDIT:
> I meant that by an historical point of view model garden was more related to Google own research papers that on third_party research.

True -- but we'll focus on things not TF2/Keras implemented yet, such as semantic segmentation, DeepLab, etc. If this is in the long run outperforming anchor-based models, (as well as DETR I guess), then we'd need to consider having re-useable layers for them.

So "long run" is the vague word here, would it make more sense to put in addons?

**Comment by bhack** [comment]
> but we'll focus on things not TF2/Keras implemented yet, such as semantic segmentation, DeepLab, etc

If you are talking about official models is true. But there are many third_party very good impl related to your list on Github.

**Comment by tanzhenyu** [comment]
> > but we'll focus on things not TF2/Keras implemented yet, such as semantic segmentation, DeepLab, etc
> 
> If you are talking about official models is true. But there are many third_party very good impl related to your list on Github.

Yes -- and I'd like to consolidate the efforts. If that's already very good impl we should ask the author to migrate directly to this repo (or Keras-applications if it's backbone)

**Comment by bhack** [comment]
@tanzhenyu nevermind anchor-free just appeared in the new model garden object detection API (Centernet) https://github.com/tensorflow/models/tree/master/research/object_detection#tensorflow-2-support

**Comment by bhack** [comment]
IMHO less reusable when we have utils in models repos https://github.com/tensorflow/models/blob/master/research/object_detection/utils/target_assigner_utils.py#L44
